### PR TITLE
Remove deprecated items from the agent

### PR DESCRIPTION
**Pull request by tannalynn**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
This PR addressed several tickets related to removing previously deprecated items in the agent.

# Related Github Issue
#479 #481 #480 #478

### CODE REVIEW

File: lib/new_relic/agent/instrumentation/active_record_notifications.rb [hunk 1]
```diff
@@ -92,22 +92,6 @@ def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name =
...
-  end
-
-  depends_on do
-    # If the deprecated :disable_active_record_5 setting is true, and
-    # the active record version is five, disable!
-    NewRelic::Agent.config[:disable_active_record_5] == false \
-        || ::ActiveRecord::VERSION::MAJOR.to_i  != 5
-  end
-
-
```
[the remark this answers is not in the document]
Comment (tannalynn) [reply lib/new_relic/agent/instrumentation/active_record_notifications.rb:110]: There is already a depends_on block checking on `disable_activerecord_instrumentation` above on line 90, so I believe we are already covered there. I think these ones were just left still to cover these deprecated config options. I think we're good to just remove this.

File: lib/new_relic/agent/new_relic_service.rb [hunk 2]
```diff
@@ -347,14 +346,6 @@ def create_and_start_http_connection
         conn = create_http_connection
         start_connection(conn)
         conn
-      rescue Timeout::Error
-        if @use_bundled_certs == false
-          ::NewRelic::Agent.logger.info("Unable to connect. Falling back to bundled security certificates")
-          @use_bundled_certs = true
-          retry
-        else
```
[the remark this answers is not in the document]
Comment (tannalynn) [reply lib/new_relic/agent/new_relic_service.rb]: That's a great idea, I went ahead and added an info log back here. Thank you!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+
+    The attribute `httpResponseCode` was previously deprecated and replaced with `http.statusCode`. This deprecated attribute has now been removed.
+
+  * **Removed deprecated option in notice_error**
+
+    Previously, the `:trace_only` option to NewRelic::Agent.notice_error was deprecated and replaced with `:expected`. This deprecated option has been removed.
+
+  * **Removed deprecated api methods**
+
+    Previously the api methods `create_distributed_trace_payload` and `accept_distributed_trace_payload` were deprecated. These have now been removed. Instead, please see `create_distributed_trace_headers` and `accept_distributed_trace_headers`, respectively.
```
Comment (mwlang) [inline CHANGELOG.md:30]: Typo in this entry.  please see `create_distributed_trace_headers` needs to be `insert_distributed_trace_headers`

Comment (tannalynn) [reply CHANGELOG.md:30]: Good catch! Thank you!

**Review by mwlang — APPROVED** [review]
LGTM.  I've reviewed this PR multiple times, and it's solid and ready to merge.

**Pull request merged**